modules/exploits/windows/local/cve_2023_21768_afd_lpe.rb

```diff
@@ -0,0 +1,102 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Post::Windows::FileInfo
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        {
+          'Name' => 'Ancillary Function Driver (AFD) for WinSock Elevation of Privilege',
+          'Description' => %q{
+            A vulnerability exists in the Windows Ancillary Function Driver for Winsock
+            (`afd.sys`) can be leveraged by an attacker to escalate privileges to those of
+            NT AUTHORITY\SYSTEM. Due to a flaw in `AfdNotifyRemoveIoCompletion`, it is
+            possible to create an arbitrary kernel Write-Where primitive, which can be used
+            to manipulate internal I/O ring structures and achieve local privilege
+            escalation.
+
+            This exploit only supports Windows 11 22H2 up to build 22621.963 (patched in
+            January 2023 updates).
+          },
+          'License' => MSF_LICENSE,
+          'Author' => [
+            'chompie', # Github PoC
+            'b33f', # Github PoC
+            'Yarden Shafir', # I/O Ring R/W primitive PoC
+            'Christophe De La Fuente' # Metasploit module
+          ],
+          'Arch' => [ ARCH_X64 ],
+          'Platform' => 'win',
+          'SessionTypes' => [ 'meterpreter' ],
+          'Privileged' => true,
+          'Targets' => [
+            [ 'Windows 11 22H2 x64', { 'Arch' => ARCH_X64 } ]
+          ],
+          'Payload' => {
+            'DisableNops' => true
+          },
+          'References' => [
+            [ 'CVE', '2023-21768' ],
+            [ 'URL', 'https://github.com/xforcered/Windows_LPE_AFD_CVE-2023-21768' ],
+            [ 'URL', 'https://github.com/yardenshafir/IoRingReadWritePrimitive' ]
+          ],
+          'DisclosureDate' => '2023-01-10',
+          'DefaultTarget' => 0,
+          'Notes' => {
+            'Stability' => [],
```

Comment:
```suggestion
            'Stability' => [ CRASH_SAFE ],
```
Would it be appropriate to add CRASH_SAFE here?

Reply:
I was hesitant to add `CRASH_SAFE` here. The exploit looks safe, I have not experienced any crash during testing. But, since it is an arbitrary write into kernel memory, I'm not 100% sure it won't crash one day. That said, if you think it is okay, I'm fine to add `CRASH_SAFE`.